src/plot/series/whisker-series.js

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2016 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import Animation from 'animation';
+import {ANIMATED_SERIES_PROPS} from 'utils/series-utils';
+import {DEFAULT_SIZE, DEFAULT_OPACITY} from 'theme';
+
+import AbstractSeries from './abstract-series';
+
+const predefinedClassName = 'rv-xy-plot__series rv-xy-plot__series--whisker';
+const DEFAULT_STROKE_WIDTH = 1;
+const DEFAULT_CROSS_BAR_WIDTH = 6;
+
+class WhiskerSeries extends AbstractSeries {
+
+  render() {
+    const {
+      animation, className, data, marginLeft, marginTop, strokeWidth, crossBarWidth, style
+    } = this.props;
+    if (!data) {
+      return null;
+    }
+    if (animation) {
+      return (
+        <Animation {...this.props} animatedProps={ANIMATED_SERIES_PROPS}>
+          <WhiskerSeries {...this.props} animation={null}/>
+        </Animation>
+      );
+    }
+
+    const sizeFunctor = this._getAttributeFunctor('size');
+    const opacityFunctor = this._getAttributeFunctor('opacity');
+    const fillFunctor = this._getAttributeFunctor('fill') ||
+      this._getAttributeFunctor('color');
+    const strokeFunctor = this._getAttributeFunctor('stroke') ||
+      this._getAttributeFunctor('color');
+    const xFunctor = this._getAttributeFunctor('x');
+    const yFunctor = this._getAttributeFunctor('y');
+
+    return (
+      <g className={`${predefinedClassName} ${className}`}
+         ref="container"
+         transform={`translate(${marginLeft},${marginTop})`}>
+        {data.map((d, i) => {
+          const r = sizeFunctor ? sizeFunctor(d) : DEFAULT_SIZE;
+          const cx = xFunctor(d);
+          const cy = yFunctor(d);
+          const positiveVariance = yFunctor(Object.assign({}, d, { y: d.y + d.variance / 2 }));
+          const negativeVariance = yFunctor(Object.assign({}, d, { y: d.y - d.variance / 2 }));
```

Comment:
do you need positive and negative variance? It seems like you could just compute the variance in coordinates and use that to translate the central y value in the object below